### PR TITLE
GSLUX-734: My symbols

### PR DESCRIPTION
**Pull request by AlitaBernachot**
### JIRA issue

https://jira.camptocamp.com/browse/GSLUX-734

### Description

Display private symbols when user is connected. Allow user to upload new private symbols.

To make it work, you must: 
- deactivate all CORS plugin
- run backend on localhost:8080 and allow all requests from vite host
- update db (symbol table and add symbols, see db repo: https://github.com/camptocamp/luxembourg_dev_db/tree/add_symbols)

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -71,6 +71,21 @@ Cypress.Commands.add('dragVertexOnMap', (originX, originY, x, y) => {
   cy.get('div.ol-viewport').realMouseUp()
 })
 
+Cypress.Commands.add('login', () => {
+  cy.intercept('POST', '/login', {
+    statusCode: 200,
+    body: {
+      login: 'MyLogin',
+      mail: 'my_login@email.com',
+    },
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply cypress/support/commands.ts:80]: I don't understand what you mean, you can use authentication anywhere in e2e tests with `cy.login()` command now.

File: src/components/draw/feature-edit-symbol-icon.vue [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+onMounted(() => {
+  // Force open list of symbol icons if feature has icon
+  if (feature.featureStyle.symbolId) {
+    currentTab.value =
+      feature.featureStyle.symboltype === 'us' ? tabs.mySymbols : tabs.publics
+  }
+})
+
+async function getSymbols(privateSymbols = false, forceRefresh = false) {
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/components/draw/feature-edit-symbol-icon.vue:67]: There is absolutely no cache anywhere in prod.

File: src/components/draw/feature-edit-symbol-icon.vue [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  />
+
+  <div class="text-right mt-3">
+    <template v-if="currentTab === tabs.mySymbols">
+      <input
+        accept="image/*"
+        ref="fileInput"
+        type="file"
+        id="file"
+        multiple
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/components/draw/feature-edit-symbol-icon.vue:150]: Good point!

File: src/services/api/api-symbols.service.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+let cachePublicSymbols: SymbolJsonResponse | undefined = undefined
+let cachePrivateSymbols: SymbolJsonResponse | undefined = undefined
+
+/**
+ * Call api to retrieve list of public symbols
+ * @param forceRefresh Ignore cache and call api, false by default
+ */
+export async function fetchPublicSymbols(forceRefresh = false) {
```
[the remark this answers is not in the document]
Comment (AlitaBernachot) [reply src/services/api/api-symbols.service.ts:27]: there is a cache for private symbols, and on for the public symbols, the only same is this line:
`const response = await fetchApi(SYMBOLS_URL, payload, 'GET')`
I don't see how you can improve readability with the conditions i will need to add.

File: src/components/draw/feature-edit-symbol-icon.vue [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  </div>
+
+  <!-- List of simple symbols -->
+  <template v-if="currentTab === tabs.configurables">
+    <div class="flex gap-1 my-2">
+      <button
+        v-for="shape in Object.keys(shapeComponents)"
+        :key="shape"
+        class="lux-btn p-2 rounded-sm border-gray-600"
+        @click="() => $emit('changeSymbolShape', shape as keyof typeof shapeComponents)"
```
[the remark this answers is not in the document]
Comment (mki-c2c) [reply src/components/draw/feature-edit-symbol-icon.vue:122]: no, no problem, I am perfectly fine with arrow functions, these are rather standard elements of JS syntax

At first sight I could not make any sense of the succession of keywords without any parentheses and commas, making it difficult to see what logically belongs to what.
Being valid TS, does not mean it is readable.
I searched in the typescript manual and maybe figured out what it means, not sure though.

I am not a TS expert, but why not make the code readable also to someone not so familiar with advanced TS functions?

For me it seems easier to understand if there is a separate type definition:
```
const shapeComponents = {
  circle: Circle,
  square: Square,
  cross: Cross,
  triangle: Triangle,
}
type shapeKey = keyof (typeof shapeComponents)

function onClick(shape: shapeKey) {
   $emit('changeSymbolShape', shape)
}
...

@click="onClick(shape as shapeKey)"
```
it's not like explaining the code in comments, but rather nice step by step programming, making clear what is done (still I am not 100% sure I have understood correctly)

**Pull request merged**